yt_dlp/extractor/iwara.py

```diff
@@ -98,10 +100,29 @@ def _real_extract(self, url):
         }
 
 
-class IwaraUserIE(InfoExtractor):
+class IwaraPlaylistBaseIE(InfoExtractor):
+    _PER_PAGE = 32
+
+    def _request_page(self, page, *args):
+        raise ExtractorError('Implement this method in subclasses')
+
+    def _entries(self, *args):
+        *args, first_page, page = args
+        videos = self._request_page(page, *args) if page or not first_page else first_page
+        for x in traverse_obj(videos, ('results', ..., 'id')):
+            yield self.url_result(f'https://iwara.tv/video/{x}')
+
+    def _paged_list(self, playlist_id, playlist_title, first_page, *eargs):
+        return self.playlist_result(
+            OnDemandPagedList(
+                functools.partial(self._entries, *eargs, first_page),
+                self._PER_PAGE),
+            playlist_id, playlist_title)
```

Comment:
Why make the signature so weird?

Reply:
They are because:
1. To allow use of fetched page 1 in some cases (`first_page`)
2. Make it work with `functools.partial` (`page`)
3. And to pass arguments to `_request_page`
4. Also to reduce duplicate code as possible